From c647acc1891528086b3b22f4bf815863f36ed90e Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 14 Apr 2018 15:24:12 +0200
Subject: [PATCH 1/1] env: mmc/fat/ext4: undefined reference to
 `mmc_initialize'

For CONFIG_ENV_FAT_INTERFACE != 'mmc' a link error
env/fat.c:93: undefined reference to `mmc_initialize'
occurs if CONFIG_MMC_SUPPORT is not enabled.

Fixes: 26862b4a40c3 ("env: mmc/fat/ext4: make sure that the MMC sub-system
is initialized before using it")
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 env/ext4.c | 2 ++
 env/fat.c  | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/env/ext4.c b/env/ext4.c
index 6c69a0a57f..a2d32b9c8d 100644
--- a/env/ext4.c
+++ b/env/ext4.c
@@ -87,8 +87,10 @@ static int env_ext4_load(void)
 	int err;
 	loff_t off;
 
+#ifdef CONFIG_MMC
 	if (!strcmp(CONFIG_ENV_EXT4_INTERFACE, "mmc"))
 		mmc_initialize(NULL);
+#endif
 
 	part = blk_get_device_part_str(CONFIG_ENV_EXT4_INTERFACE,
 				       CONFIG_ENV_EXT4_DEVICE_AND_PART,
diff --git a/env/fat.c b/env/fat.c
index 19f260e881..884eeb8c5f 100644
--- a/env/fat.c
+++ b/env/fat.c
@@ -89,8 +89,10 @@ static int env_fat_load(void)
 	int dev, part;
 	int err;
 
+#ifdef CONFIG_MMC
 	if (!strcmp(CONFIG_ENV_FAT_INTERFACE, "mmc"))
 		mmc_initialize(NULL);
+#endif
 
 	part = blk_get_device_part_str(CONFIG_ENV_FAT_INTERFACE,
 					CONFIG_ENV_FAT_DEVICE_AND_PART,
-- 
2.17.0

